src/main/java/net/snowflake/client/jdbc/SnowflakeBasicDataSource.java

```diff
@@ -85,9 +85,11 @@ public Connection getConnection(String username, String password) throws SQLExce
     if (!AUTHENTICATOR_OAUTH.equalsIgnoreCase(
         authenticator)) { // For OAuth, no username is required
       properties.put("user", username);
-      if (!AUTHENTICATOR_SNOWFLAKE_JWT.equalsIgnoreCase(authenticator)) {
-        properties.put("password", password);
-      }
+    }
+
+    // The driver needs password for OAUTH as part of SNOW-533673 feature request.
```

Comment:
nit: The driver _might_ need the password field for OAuth as part of SNOW-533673 feature request.

Reply:
Done. Checks during open session as other params checks happen at same place.